tests/SideBySide.New/BulkLoaderAsync.cs

```diff
@@ -0,0 +1,507 @@
+					bl.NumberOfLinesToSkip = 1;
+					bl.Expressions.Add("five = UNHEX(five)");
+					bl.LineTerminator = "\\n";
+					bl.FieldTerminator = "\\t";
+					bl.Local = true;
+					int rowCount = await bl.LoadAsync();
+					Assert.Equal(20, rowCount);
+				}
+			}
+			finally
+			{
+				await FinalizeTestAsync();
+			}
+		}
+		[Fact]
+		public async Task BulkLoadCsvFile()
+		{
+			try
+			{
+				await InitializeTestAsync();
+
+				using (MySqlConnection connection = new MySqlConnection(AppConfig.ConnectionString))
+				{
+					MySqlBulkLoader bl = new MySqlBulkLoader(connection);
+					bl.FileName = AppConfig.MySqlBulkLoaderCsvFile;
+					bl.TableName = m_testTable;
+					bl.Columns.AddRange(new string[] { "one", "two", "three", "four", "five" });
+					bl.NumberOfLinesToSkip = 1;
+					bl.FieldTerminator = ",";
+					bl.FieldQuotationCharacter = '"';
+					bl.FieldQuotationOptional = true;
+					bl.Expressions.Add("five = UNHEX(five)");
+					bl.Local = false;
+					int rowCount = await bl.LoadAsync();
+					Assert.Equal(20, rowCount);
+				}
+			}
+			finally
+			{
+				await FinalizeTestAsync();
+			}
+		}
+		[Fact]
+		public async Task BulkLoadLocalCsvFile()
+		{
+			try
+			{
+				await InitializeTestAsync();
+
+				MySqlBulkLoader bl = new MySqlBulkLoader(m_database.Connection);
+				bl.FileName = AppConfig.MySqlBulkLoaderLocalCsvFile;
+				bl.TableName = m_testTable;
+				bl.CharacterSet = "UTF8";
+				bl.Columns.AddRange(new string[] { "one", "two", "three", "four", "five" });
+				bl.NumberOfLinesToSkip = 1;
+				bl.FieldTerminator = ",";
+				bl.FieldQuotationCharacter = '"';
+				bl.FieldQuotationOptional = true;
+				bl.Expressions.Add("five = UNHEX(five)");
+				bl.Local = true;
+				int rowCount = await bl.LoadAsync();
+				Assert.Equal(20, rowCount);
+			}
+			finally
+			{
+				await FinalizeTestAsync();
+			}
+		}
+		[Fact]
+		public async Task BulkLoadCsvFileNotFound()
+		{
+			try
+			{
+				await InitializeTestAsync();
+
+				MySqlBulkLoader bl = new MySqlBulkLoader(m_database.Connection);
+				bl.FileName = AppConfig.MySqlBulkLoaderCsvFile + "-junk";
+				bl.TableName = m_testTable;
+				bl.CharacterSet = "UTF8";
+				bl.Columns.AddRange(new string[] { "one", "two", "three", "four", "five" });
+				bl.NumberOfLinesToSkip = 1;
+				bl.FieldTerminator = ",";
+				bl.FieldQuotationCharacter = '"';
+				bl.FieldQuotationOptional = true;
+				bl.Expressions.Add("five = UNHEX(five)");
+				bl.Local = false;
+				try
+				{
+ 					int rowCount = await bl.LoadAsync();
+				}
+				catch (MySqlException mySqlException)
+				{
+					if (mySqlException.InnerException != null)
+					{
+						Assert.IsType(typeof(System.IO.FileNotFoundException), mySqlException.InnerException);
+					}
+					else
+					{
+						Assert.Contains("Errcode: 2 - No such file or directory", mySqlException.Message);
+					}
+				}
+				catch (System.InvalidOperationException invalidOperationException)
```

Comment:
Note that this `catch` isn't in [`BulkLoaderSync.cs`](https://github.com/mysql-net/MySqlConnector/pull/156/files#diff-a5778fc2df515d66b90430cea4618cb2R170).